src/os/filestore/FileStore.cc

```diff
@@ -231,16 +231,24 @@ int FileStore::lfn_open(const coll_t& cid,
 			const ghobject_t& oid,
 			bool create,
 			FDRef *outfd,
-                        Index *index)
+			Index *index,
+			bool directio)
 {
   assert(outfd);
   int r = 0;
   bool need_lock = true;
   int flags = O_RDWR;
+  FDCache *cachep;
```

Comment:
At least  move this adjacent to the block which actually sets it.  I'd rather you actually did

```c++
FDCache *cachep = directio ? &fdcache_direct : &fdcache;
````

That would reduce the likelihood of someone later using it before it's initialized.